src/components/rainbow-toast/getToastFromTransaction.ts

```diff
@@ -0,0 +1,104 @@
+import { type RainbowToast } from '@/components/rainbow-toast/types';
+import { RainbowTransaction } from '@/entities';
+import { Navigation } from '@/navigation';
+import Routes from '@/navigation/routesNames';
+import { Mints } from '@/resources/mints';
+
+export const txIdToToastId = (tx: RainbowTransaction): string => (tx.nonce ? `${tx.nonce}` : tx.hash + (tx.chainId || tx.asset?.chainId));
```

Comment:
leaving a comment to revisit this - we need to handle scenarios of switching wallets